Documentation/Browser/LibJSGarbageCollection.md

```diff
@@ -0,0 +1,27 @@
+# LibJS Garbage Collection
+
+LibJS has a [non-generational](https://en.wikipedia.org/wiki/Tracing_garbage_collection#Generational_GC_(ephemeral_GC)), [non-compacting](https://en.wikipedia.org/wiki/Tracing_garbage_collection#Moving_vs._non-moving), [stop-the-world](https://en.wikipedia.org/wiki/Tracing_garbage_collection#Stop-the-world_vs._incremental_vs._concurrent) [mark-and-sweep](https://en.wikipedia.org/wiki/Tracing_garbage_collection#Na%C3%AFve_mark-and-sweep) garbage collector.
+
+## Allocating Cells
+
+Allocation and garbage collection are tightly integrated and therefore the story of a JS object and it's relationship with the GC starts at it's allocation. The `Interpreter` has a member of the `Heap` class which allows to allocate a new `Cell` which will kept alive as long as the Interpreter needs it (has an active reference to it). While the GC is a integral part of the JS engine, it is not limited to just JS objects. Any class like the `Object` class (JS object) can inherent from `Cell` and can than be managed by the GC. The browser for example also leverages the GC to manage DOM element.
+
+The actual allocations of the objects aren't made with `malloc` and `free` but are made inside `HeapBlock`s of 16KB each. This is an performance improvement as the GC has to visit the entire heap during collection and this increases locality (without being a compacting GC).
+
+The `Heap` class outsources the allocations to the `CellAllocator` which manages a list of heap blocks. Cells can have different size's because different JS objects inherit it. Therefore,  multiple cell allocators exist where each manages cells of a particular maximum size and the smallest allocator that can hold the required cell is chosen
```

Comment:
```suggestion
The `Heap` class outsources the allocations to the `CellAllocator` which manages a list of heap blocks. Cells can have different sizes because different JS objects inherit from `Cell`. Therefore,  multiple cell allocators exist where each one manages cells of a particular maximum size and the smallest allocator that can hold the required cell is chosen.
```